File: Codes/3350-adjacent-increasing-subarrays-detection-ii.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-10-15 22:07:17
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-10-15 22:12:42
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

// THIS IS NOT RIGHT
class Solution {
public:
    int maxIncreasingSubarrays(vector<int>& nums) {
        int ans = 1;
        int lastVal = 1000000001, lastCnt = 0, nowCnt = 0;
        for (int t : nums) {
            if (t <= lastVal) {
                ans = max(ans, min(lastCnt, nowCnt));
                ans = max(ans, nowCnt / 2);
                lastCnt = nowCnt, nowCnt = 0;
            } else {
                nowCnt++;
            }
            lastVal = t;
        }
        return ans;
    }
};

File: Codes/3350-adjacent-increasing-subarrays-detection-ii.rs
/*
 * @Author: LetMeFly
 * @Date: 2025-10-15 22:07:17
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-10-15 22:32:24
 */
impl Solution {
    pub fn max_increasing_subarrays(nums: Vec<i32>) -> i32 {
        let mut ans: i32 = 0;
        let mut last_cnt: i32 = 0;
        let mut now_cnt: i32 = 0;
        for i in 0..nums.len() {
            now_cnt += 1;
            if i == nums.len() - 1 || nums[i] >= nums[i + 1] {
                ans = ans.max(last_cnt.min(now_cnt)).max(now_cnt / 2);
                // (last_cnt, now_cnt) = (now_cnt, 0);
                last_cnt = now_cnt;
                now_cnt = 0;
            }
        }
        ans
    }
}